applications/nrf_desktop/configuration/nrf52840_pca10059/ble_scan_def.h

```diff
@@ -13,5 +13,21 @@
  */
 const struct {} ble_scan_include_once;
 
+#include "ble_event.h"
+
 /* Filters used for scanning */
-char *bt_peripherals[] = { "Mouse nRF52", "Kbd nRF52" };
+struct bt_peripheral_scan_filter {
+	char *short_name;
+	u8_t peer_type_bm;
+};
+
+static const struct bt_peripheral_scan_filter bt_peripheral_scan_filters[] = {
+	{
+		.short_name = "Mouse nRF52",
+		.peer_type_bm = (1 << PEER_TYPE_POS_MOUSE),
```

Comment:
why do we need to separately define `peer_type_bm`?

maybe this would do:
```
static const char *peer_type_short_name[] = {
   [PEER_TYPE_MOUSE] = "Mouse ...",
   [PEER_TYPE_KEYBOARD] = "Kbd ..."
};
```

and when you need a bitmask you do `BIT(peer_type)` or `BIT(PEER_TYPE_MOUSE)`?

Reply:
I wanted to give more flexibility, e.g one device could be mouse and keyboard at once.
Additionaly even now there is more than one type of mouse (there could be 2 mice with various short names).